discovery-provider/src/tasks/calculate_trending_challenges.py

```diff
@@ -32,15 +32,21 @@ def date_to_week(date: datetime) -> str:
     return date.strftime("%Y-%m-%d")
 
 
-def get_latest_blocknumber(session: Session, redis: Redis) -> Optional[int]:
+def get_latest_blocknumber_via_redis(session: Session, redis: Redis) -> Optional[int]:
     # get latest db state from redis cache
     latest_indexed_block_num = redis.get(most_recent_indexed_block_redis_key)
     if latest_indexed_block_num is not None:
         return int(latest_indexed_block_num)
+
+    return get_latest_blocknumber(session)
+
+
+def get_latest_blocknumber(session: Session) -> Optional[int]:
```

Comment:
i know i was importing this function for aggregate_user but maybe we should just move this there? seems strange to modify trending just to make this simple query that trending doesn't even use.